code/obj/flock/structure/rift.dm

```diff
@@ -45,7 +43,10 @@
 					candidate_turfs -= S
 					continue
 				if(prob(25))
-					src.flock.claimTurf(flock_convert_turf(S))
+					if (src.flock)
```

Comment:
what's this for?

Reply:
If you suicide while you have a rift

Reply:
if you suicide with a rift, it should kill the rift.

Reply:
Unless there's a specific reason for deleting the rift, I think it could be done either way?

Reply:
I say delete the rift because without a flockmind, there isn't really a flock to spawn. I guess you could do something like triggering the partition ability and asking for a ghost to take on the role when the egg pops.

Reply:
resolved, by medass

Reply:
![image](https://user-images.githubusercontent.com/65367576/169670122-5679e4a9-9c6a-4fba-8194-15b6d233a68a.png)